state/cache.go

```diff
@@ -64,8 +64,13 @@ func (s *CacheState) RefreshState() error {
 		// Cache is newer than remote. Not a big deal, user can just
 		// persist to get correct state.
 		s.refreshResult = CacheRefreshLocalNewer
-	case cached == nil && durable != nil:
+	case !cached.HasResources() && durable != nil:
```

Comment:
We should probably verify that cached is not nil here since the `HasResources()` method accesses members.